roles/pulp_health_check/tasks/main.yml

```diff
@@ -8,18 +8,39 @@
   debug:
     msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulpcore")|list|items2dict }}'
 
-
 - name: Checking Pulp services
   fail:
     msg: "{{ item.value.name }} state: {{ item.value.state }}"
   when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
   with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulpcore")|list|items2dict }}'
 
+# Previous approach to handle the delay, currently using the alternating timeout approach instead.
+#
+# Without this separate task vs the one below, the default timeout of 30 seconds
+# 6 seconds delay can make users wait for 1.5 hours (with 150 retries.)
+#- name: Wait for Pulp to respond to HTTP requests at all
+#  uri:
+#    url: "http://{{ __pulp_default_host if pulp_api_bind.startswith('unix:') else pulp_api_bind }}/pulp/api/v3/status/"
+#    status_code: "200"
+#    unix_socket: "{{ pulp_api_bind | regex_replace('^unix:', '') if pulp_api_bind.startswith('unix:') else omit }}"
+#    timeout: 30
+#  #failed_when: result.msg is search('timed out')
+#  register: result
+#  until: >
+#    result.msg is defined and
+#    result.msg is not search('timed out')
+#  delay: 6
+#  retries: 18
+#  check_mode: false
+
 - name: Ensure Pulp is up and healthy
   uri:
     url: "http://{{ __pulp_default_host if pulp_api_bind.startswith('unix:') else pulp_api_bind }}/pulp/api/v3/status/"
     status_code: "200"
     unix_socket: "{{ pulp_api_bind | regex_replace('^unix:', '') if pulp_api_bind.startswith('unix:') else omit }}"
+    # Alternate between 30 second timeouts and 5 second timeouts when handling the situation of
```

Comment:
why these magic numbers?

Reply:
30 is the default. It handles hypothetical extremely long times for the webserver to produce the page.

We intermix it with 5 second waits, because if the server isn't up yet and suddenly is, a 5 second wait usually is sufficient I would think.

The point is to lower the total amount of time that the CI or the user may wait on the health check.

Reply:
it was more from where they came from, thank you for all the explanation